File: Microservices/IoT.House.Automation.Microservices.Arduino/IoT.House.Automation.Microservices.Arduino.Infra/RabbitMQ/Config/RabbitMQConfig.cs
using System;
using System.Collections.Generic;
using System.Text;
using IoT.House.Automation.Libraries.ConfigLoader.Abstractions;

namespace IoT.House.Automation.Microservices.Arduino.Infra.RabbitMQ.Config
{
    public class RabbitMQConfig : BaseConfigLoader
    {
        public string Uri { get; set; }
        public string Exchange { get; set; }
        public string ExchangeType { get; set; }

        protected override string ProjectName => GetProjectName();
        protected override string ClassName => GetType().Name;

        public RabbitMQConfig(IConfigLoaderRepository loaderRepository) : base(loaderRepository)
        {
            TriggerLoadingConfig(this);
        }

        private string GetProjectName() => "IoT.House.Automation";
    }
}

File: Microservices/IoT.House.Automation.Microservices.Mobile/infrastructure/src/main/java/iot/house/automation/microservices/mobile/infra/rabbitMQ/config/RabbitMQConfig.java
package iot.house.automation.microservices.mobile.infra.rabbitMQ.config;

import lombok.Getter;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

@Component
@Getter
public class RabbitMQConfig {
    private final String uri;
    private final String publishExchange;
    private final String listenerExchange;
    private final String exchangeType;
    private int retryCount;

    public RabbitMQConfig(@Value("${spring.rabbitmq.uri}") String uri,
                          @Value("${spring.rabbitmq.publisherExchange}") String publishExchange,
                          @Value("${spring.rabbitmq.listenerExchange}") String listenerExchange,
                          @Value("${spring.rabbitmq.exchangeType}") String exchangeType,
                          @Value("${spring.rabbitmq.retryCount}") int retryCount)
    {
        this.uri = uri;
        this.publishExchange = publishExchange;
        this.listenerExchange = listenerExchange;
        this.exchangeType = exchangeType;
        this.retryCount = retryCount;
    }
}
